### PR TITLE
fix: disable last modified check on vercel edge config and fix type definitions

### PR DESCRIPTION
**Pull request by ajwootto**
- make sure edge config always returns a config in Nextjs by disabling the last modified check in that SDK
- fix the type definitions so that the edge config client is compatible with Nextjs types

### CODE REVIEW

File: lib/shared/vercel-edge-config/src/edge-config.ts [hunk 1]
```diff
@@ -7,12 +7,14 @@ export class EdgeConfigSource extends ConfigSource {
         super()
     }
 
-    async getConfig(
+    async getConfig<T extends boolean>(
         sdkKey: string,
         kind: 'server' | 'bootstrap',
         obfuscated: boolean,
+        lastModifiedThreshold?: string,
```
Comment (kaushalkapasi) [inline lib/shared/vercel-edge-config/src/edge-config.ts:14]: where's this used?

Comment (ajwootto) [reply lib/shared/vercel-edge-config/src/edge-config.ts:14]: it's not used here but its part of the type signature for the abstract class

File: sdk/nextjs/src/pages/bucketing.ts [hunk 2]
```diff
@@ -49,6 +48,8 @@ export const getBucketedConfig = async (
         sdkKey,
         'bootstrap',
         obfuscated,
+        '',
```
Comment (kaushalkapasi) [inline sdk/nextjs/src/pages/bucketing.ts:51]: so we're no longer allowing the `lastModifiedThreshold` to be passed in?

Comment (ajwootto) [reply sdk/nextjs/src/pages/bucketing.ts:51]: it never was

**Review by kaushalkapasi — APPROVED** [review]

**Pull request merged**